themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl

```diff
@@ -27,27 +27,28 @@
     <div class="product-cover">
       {if $product.default_image}
         <img
-          class="js-qv-product-cover"
-          src="{$product.default_image.bySize.large_default.url}"
+          class="js-qv-product-cover img-fluid"
+          src="{$product.default_image.bySize.medium_default.url}"
```

Comment:
One last question, why not keeping the large image (using the large size in the image attributes) but this should now display correctly thanks to the `img-fluid` no? This way it ensures us to have a good quality

Reply:
@jolelievre Because then you are loading an image 4x bigger for nothing. It's selfish to load 200kB image instead of 50 kB only because it looks better on a Mac. All other users would only experience slow down in page load, lower score in Google pagespeed and worse SEO.

To benefit of increased resolutions on scaled displays, you should use srcsets. On anything above 100%, the browser will choose the bigger photo. Or not, it can maybe even adjust it's behavior on low-data-consumption modes.

`<img src="/photo.jpg" srcset="/photo_2timesbigger.jpg 2x">`